File: 0771-jewels-and-stones/0771-jewels-and-stones.js
const numJewelsInStones = (jewels, stones) => {
  let total = 0;

  const map = new Map();

  for (const jewel of jewels) {
    const count = map.get(jewel) ?? 0;
    map.set(jewel, count + 1);
  }

  for (const stone of stones) {
    if (map.has(stone)) {
      total++;
    }
  }

  return total;
};

File: 0771-jewels-and-stones/0771-jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        total = 0
        
        pattern = set()
        
        for jewel in jewels:
            pattern.add(jewel)
        
        for stone in stones:
            if stone in pattern:
                total += 1
            
        return total
    